main.rb

```diff
@@ -0,0 +1,133 @@
+# Recreation of enumerables.
+# rubocop: disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Style/CaseEquality
+module Enumerable
+  def my_each
+    return enum_for(__method__) unless block_given?
+
+    size.times { |i| yield to_a[i] }
+    self
+  end
+
+  def my_each_with_index
+    return enum_for(__method__) unless block_given?
+
+    size.times { |i| yield to_a[i], i }
+    self
+  end
+
+  def my_select
+    return enum_for(__method__) unless block_given?
+
+    array = []
+    my_each { |n| array << n if yield n }
+    array
+  end
+
+  def my_all?(arg = nil)
+    puts "#{__FILE__}:#{__LINE__}: w: block not used" if arg && block_given?
+    return my_all_extra_func(arg) if arg
+
+    if !block_given?
+      size.times { |n| return false if to_a[n] == false || to_a[n].nil? }
+    else
+      my_each { |n| return false unless yield n }
+    end
+    true
+  end
+
+  def my_all_extra_func(arg)
+    if arg.is_a?(Regexp)
+      my_each { |n| (n =~ arg).nil? ? false : 'Error' }
+      true
+    elsif arg
+      my_each { |n| return false unless n.is_a? arg }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if arg
+      puts "#{__FILE__}:#{__LINE__}: w: block not used" if block_given?
+      return my_any_extra_func(arg)
+    elsif block_given?
+      my_each { |n| return true if yield n }
+    else
+      my_each { |n| return true if n }
+    end
+    false
+  end
```

Comment:
# my_any?
1. [ ] When a class is passed as an argument returns true if at least one of the collection is a member of such class:: Numeric

### Test 👉  `array.my_any?(Numeric)`

|Expected|Got|
|:---:|:---:|
|true|`#<Array:2080> => [2, 4, 7, 3, 7, 1, 3, 5, 7, 7, 4, 4, 2, 7, 3, 3, 4, 2, 5, 2, 1, 3, 4, 5, 8, 2, 6, 0, 5, 0, 0, 4, 5, 1..., 5, 8, 4, 5, 2, 4, 8, 2, 3, 3, 4, 5, 6, 0, 4, 5, 7, 3, 3, 2, 6, 0, 3, 1, 8, 1, 1, 2, 1, 8, 8, 6, 0]`|

#

2. [ ] When a Regex is passed as an argument returns true if any of the collection matches the Regex

### Test 👉 ` words.my_any?(/d/)`

|Expected|Got|
|:---:|:---:|
|true|`#<Array:2100> => ["dog", "door", "rod", "blade"]`|

#
3. [ ]  When a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern
### Test 👉  `words.my_any?('cat')`

|Expected|Got|
|:---:|:---:|
|true|false|